File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.js
function solution(babbling) {
    var answer = 0;
    const can = ["aya", "ye", "woo", "ma"];
    babbling.forEach((word) => {
        let wordString = word;
        let canArr = [...can];
        let flag = true;
        let cur = ''
        const flag2 = canArr.some(value => {
            if(wordString.includes(value + value)){
                    return true;
                }
            return false;
        })
        while(flag){
            flag = canArr.some(value => {
                
                if(wordString.includes(value)){
                    console.log(wordString)
                    wordString = wordString.replace(value, ' ');
                    cur = value;
                    return true
                }
                return false;
            })
            // canArr = [...can];
            // canArr = canArr.filter(value => value !==cur)
        }
        console.log(flag2)
        if(wordString.trim().length === 0 && !flag2){
            answer++;
        }
    })
    console.log(answer)
    return answer;
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.py
def solution(babbling):
    words = ["aya", "ye", "woo", "ma"]
    answer = 0

    for b in babbling:
        i = 0
        prev = ""
        ok = True

        while i < len(b):
            matched = False

            for w in words:
                if b.startswith(w, i) and w != prev:
                    prev = w
                    i += len(w)
                    matched = True
                    break

            if not matched:
                ok = False
                break

        if ok:
            answer += 1

    return answer